Tribler/Test/Community/Multichain/test_multichain_community.py

```diff
@@ -558,34 +512,100 @@ def test_get_statistics_with_previous_block(self):
+        count = node.community.persistence.execute(u"SELECT COUNT(*) FROM multi_chain").fetchone()[0]
+        assert count == amount, "Wrong number of blocks in database, was {0} but expected {1}".format(count, amount)
 
     @blocking_call_on_reactor_thread
     def assertBlocksAreEqual(self, node, other):
-        ids_node = node.community.persistence.get_all_hash_requester()
-        ids_other = other.community.persistence.get_all_hash_requester()
-        assert len(ids_node) == len(ids_other)
-        blocks_node = map(node.community.persistence.get_by_hash_requester, ids_node)
-        blocks_other = map(other.community.persistence.get_by_hash_requester, ids_other)
-
-        for block_node, block_other in zip(blocks_node, blocks_other):
-            assert block_node.hash_requester == block_other.hash_requester
-            assert block_node.hash_responder == block_other.hash_responder
+        map(self.assertEqual_block,
+            node.community.persistence.crawl(node.community.my_member.public_key, 0),
+            other.community.persistence.crawl(node.community.my_member.public_key, 0))
+        map(self.assertEqual_block,
+            node.community.persistence.crawl(other.community.my_member.public_key, 0),
+            other.community.persistence.crawl(other.community.my_member.public_key, 0))
+
+    @staticmethod
+    def set_expectation(node, req, up, down):
+        if node.community.pending_bytes.get(req.community.my_member.public_key):
+            node.community.pending_bytes[req.community.my_member.public_key].add(down, up)
+        else:
+            node.community.pending_bytes[req.community.my_member.public_key] = PendingBytes(down, up)
+
+    @staticmethod
+    def create_block(req, resp, target_resp, up, down):
+        TestMultiChainCommunity.set_expectation(resp, req, up, down)
+        req.call(req.community.sign_block, target_resp, resp.my_member.public_key, up, down)
+
+        # Process packets until there are no more to process. This should give enough margin to allow for a crawl
+        # during block creation + retry of block signing.
+        rounds = 13
+        while rounds > 0:
+            responder_packets = resp.process_packets(timeout=0.2)
+            requester_packets = req.process_packets(timeout=0.2)
+            if len(responder_packets if responder_packets else []) + \
+                    len(requester_packets if requester_packets else []) == 0:
+                rounds -= 1
+
+    @staticmethod
+    def crawl_node(crawler, crawlee, target_to_crawlee, sequence_number=None):
+        crawler.call(crawler.community.send_crawl_request,
+                     target_to_crawlee, crawlee.my_member.public_key, sequence_number)
+        crawlee.give_message(crawlee.receive_message(names=[CRAWL]).next()[1], crawler)
+        TestMultiChainCommunity.transport_halfblocks(crawlee, crawler)
+
+    @staticmethod
+    def transport_halfblocks(source, sink):
```

Comment:
I find the `sink` as variable name quite vague; please use `destination` or something similar instead.